src/AppBundle/Controller/BlogController.php

```diff
@@ -114,4 +116,47 @@ public function commentFormAction(Post $post)
             'form' => $form->createView(),
         ));
     }
+
+    /**
+     * @Route("/search", name="blog_search")
+     * @Method("GET")
+     *
+     * @return JsonResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $query = $request->query->get('q', '');
+
+        // Sanitizing the query: removes all non-alphanumeric characters except whitespaces
+        $query = preg_replace('/[^[:alnum:] ]/', '', trim(preg_replace('/[[:space:]]+/', ' ', $query)));
+
+        // Splits the query into terms and removes all terms which
+        // length is less than 2
+        $terms = array_unique(explode(' ', strtolower($query)));
+        $terms = array_filter($terms, function($term) {
+            return 2 <= strlen($term);
+        });
+
+        $posts = new ArrayCollection();
```

Comment:
this should be an array. Doctrine does not return collection objects when doing queries